Replaces QuickCloudAPI with acquia-sdk-ruby

jacobbednarz opened the pull request:
This substitutes in [acquia-sdk-ruby](https://github.com/jacobbednarz/acquia-sdk-ruby) for QuickCloudAPI as it is a more robust and maintainable solution. In the future, the acquia-sdk-ruby will extend to cover the options available via the API (including logstream) as well as some system integrations.

Other minor changes:
- Bumped JSON to be 1.8.
- Minor formatting changes
- Rescuing `StandardError` instead of class specific exception

What are your thoughts on this change?

jacobbednarz commented:
closing as this hasn't been accepted - feel free to revive should you want to continue with this one.

/cc @bjaspan @rb2k

rb2k commented:
I'm afraid @bjaspan will have to look at this one, I don't have merge access anymore (at new company) :)

Looks like a useful change though. It seems to have conflicts with the master branch though?

jacobbednarz commented:
> I'm afraid @bjaspan will have to look at this one, I don't have merge access anymore (at new company) :)

Congrats! I've also moved on to another company and just looking to clean up a couple of outstanding pull requests for some third party libraries (hence this bump :stuck_out_tongue:). 

> Looks like a useful change though.

I thought so - the initial way this was done was quite dirty and fragile. I put the [acquia-sdk-ruby](https://github.com/jacobbednarz/acquia-sdk-ruby) project together as there wasn't any support for it already and I found a couple of tools which were using ruby to interact with the cloud API but doing it in odd ways so thought it would be better to consolidate and do it nicely.

I'm not too fussed either way but with tools like logtsream emerging it might be something worth while to utilise a decent base project to save time and effort.

>  It seems to have conflicts with the master branch though?

Just updated from the upstream master. Looks like a JSON version was conflicting.
